server/utils/verifiers/VerifierBase.ts

```diff
@@ -0,0 +1,350 @@
+            LOG.error(`could not find enumerator for identifier type (${identifier.idVIdentifierType}) in Cache`, LOG.LS.eTEST);
+            return null;
+        }
+
+        return { identifier, identifierType, identifierTypeEnum };
+    }
+
+    protected async getSubjectIdentifiers(subject: DBAPI.Subject, systemObject: DBAPI.SystemObject): Promise<IdentifierList | null> {
+
+        // structure to hold our results
+        const result: IdentifierList = { preferred: null, edan: null, ark: null, details: [] };
+
+        // grab the preferred identifier, if nothing then leave null so calling can decide action
+        if(subject.idIdentifierPreferred) {
+            const preferredIdentifier: DBAPI.Identifier | null = await DBAPI.Identifier.fetch(subject.idIdentifierPreferred);
+            if(!preferredIdentifier){
+                LOG.error(`subject's preferredId not found in the DB (id: ${subject.idSubject} | subject: ${subject.Name})`, LOG.LS.eTEST);
+                // subjectStats[i].isValid = false;
+            } else {
+                // grab our identifier details (type) and store it
+                const preferredIdentifierDetails: IdentifierDetails | null = await this.getIdentifierType(preferredIdentifier);
+                if(preferredIdentifierDetails) {
+                    result.preferred = preferredIdentifierDetails;
+                    result.details?.push(preferredIdentifierDetails);
+                }
+
+                // check for edan/ark and store as appropriate because the preferred id may be ARK too
+                switch(preferredIdentifierDetails?.identifierTypeEnum){
+                    case COMMON.eVocabularyID.eIdentifierIdentifierTypeEdanRecordID: {
+                        if(!result.edan) result.edan = preferredIdentifierDetails;
+                    } break;
+
+                    case COMMON.eVocabularyID.eIdentifierIdentifierTypeARK: {
+                        if(!result.ark) result.ark = preferredIdentifierDetails;
+                    } break;
+                }
+            }
+        }
+
+        // grab our list of identifiers from the SystemObject id
+        const identifiers: DBAPI.Identifier[] | null = await DBAPI.Identifier.fetchFromSystemObject(systemObject.idSystemObject);
+        if(!identifiers){
+            LOG.error(`could not get identifiers from subject (id: ${subject.idSubject} | subject: ${subject.Name})`, LOG.LS.eTEST);
+            return null;
+        }
+        if(identifiers.length<=0){
+            LOG.info(`(WARNING) no identifiers assigned to subject (id: ${subject.idSubject} | subject: ${subject.Name})`, LOG.LS.eTEST);
+            return result;
+        }
+
+        // cycle through all identifiers, find EDAN/ARK, and push to list
+        for(const identifier of identifiers) {
+
+            // get our details for this identifier, skip if error, store if valid
+            const details = await this.getIdentifierType(identifier);

```

Comment:
As a reader of this code, I find it helpful to tell me what the type of `details` is.  Of course, it's not necessary ... and it's more verbose (both to read and type).  But my religious conviction here is that it makes the code more maintainable.